code/datums/black_market_item.dm

```diff
@@ -0,0 +1,174 @@
+				continue
+			if(I.only_on_month)
+				if(time2text(world.realtime,"MM") != I.only_on_month)
+					continue
+			if(I.only_on_day)
+				if(time2text(world.realtime,"DD") != I.only_on_day)
+					continue
+
+			if(!black_market_items[I.category])
+				black_market_items[I.category] = list()
+
+			black_market_items[I.category] += I
+
+	return black_market_items
+
+// You can change the order of the list by putting datums before/after one another OR
+// you can use the last variable to make sure it appears last, well have the category appear last.
+
+/datum/black_market_item
+	var/name = "item name"
+	var/category = "item category"
+	var/desc = "item description"
+	var/item = null
+	var/num_in_stock = 0	// The stock avaliable. -1 is infinite if stock_deviation is 0.
+	var/stock_deviation = 0   //The range of stock RNG. 2 deviation with stock 5 will be anywhere from 3-7.
+	var/cost = 0
+	var/cost_deviation        //Same as stock deviation, but for cost.
+	var/display_chance = 0
+	var/abstract = 0 //Indicates if datum is to actually be displayed
+
+	var/only_on_month	//two-digit month as string
+	var/only_on_day		//two-digit day as string
+	var/static/round_stock                         //God I love if statements
+	var/static/round_stock_calculated = 0          //Allows for round-to-round variance instead of changing every time you open it
+	var/static/round_cost
+	var/static/round_cost_calculated = 0
+	var/static/active_this_round = 0
+	var/static/active_this_round_calculated = 0
+	
+/datum/black_market_item/proc/is_active()
+	if(!active_this_round_calculated)
+		if(rand(0,100) <= display_chance)
+			active_this_round = 1
+		active_this_round_calculated = 1
+	. = active_this_round
+	
+/datum/black_market_item/proc/get_cost(var/cost_modifier = 1)
+	if(!round_cost_calculated)
+		round_cost = rand(cost-cost_deviation,cost+cost_deviation)
+		round_cost_calculated = 1
+	. = Ceiling(round_cost * cost_modifier)
+
+/datum/black_market_item/proc/get_stock()
+	if(!round_stock_calculated)
+		round_stock = rand(num_in_stock-stock_deviation,num_in_stock+stock_deviation)
+		round_stock_calculated = 1
+	. = round_stock
+
+	
+/datum/black_market_item/proc/spawn_item(var/turf/loc, var/obj/item/device/illegalradio/U, mob/user)
+	U.money_stored -= get_cost()
+	feedback_add_details("traitor_black_market_items_bought", name)
+	message_admins("[key_name(user)] just purchased the [src.name] from the black market. ([formatJumpTo(get_turf(U))])")
+	return new item(loc,user)
+
+/datum/black_market_item/proc/buy(var/obj/item/device/illegalradio/U, var/mob/user)
+	..()
+	if(!istype(U)) 
+		return 0
+
+	if(user.stat || user.restrained())
+		return 0
+
+	if(!(istype(user,/mob/living/carbon/human)))
+		return 0
+
+	if(get_stock() <= 0)
+		to_chat(user, "<span class='warning'>This item is out of stock. Scram.</span>")
+		return 0
+
+	// If the black_market's holder is in the user's content
+	if((U.loc in user.contents) || (in_range(U.loc, user) && istype(U.loc.loc, /turf)))
```

Comment:
just use `getTurf(U)` instead of that `U.loc.loc` fuckery

Reply:
That's not what this is doing